opm/parser/eclipse/EclipseState/Schedule/Compsegs.cpp

```diff
@@ -0,0 +1,136 @@
+/*
+  Copyright 2015 SINTEF ICT, Applied Mathematics.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <opm/parser/eclipse/EclipseState/Schedule/Compsegs.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/ScheduleEnums.hpp>
+#include <opm/parser/eclipse/EclipseState/Grid/EclipseGrid.hpp>
+
+namespace Opm {
+
+
+    Compsegs::Compsegs(int i_in, int j_in, int k_in, int branch_number_in, double distance_start_in, double distance_end_in,
+                       WellCompletion::DirectionEnum dir_in, double center_depth_in, double thermal_length_in, int segment_number_in)
+    : m_i(i_in),
+      m_j(j_in),
+      m_k(k_in),
+      m_branch_number(branch_number_in),
+      m_distance_start(distance_start_in),
+      m_distance_end(distance_end_in),
+      m_dir(dir_in),
+      m_center_depth(center_depth_in),
+      m_thermal_length(thermal_length_in),
+      m_segment_number(segment_number_in)
+    {
+    }
+
+    std::vector<std::shared_ptr<Compsegs>> Compsegs::compsegsFromCOMPSEGSKeyword(DeckKeywordConstPtr compsegsKeyword,
```

Comment:
Maybe the `Compsegs` constructor could take a `DeckKeyword` - so that all of this decoding was in that class, and not in the general well class?